### PR TITLE
pytest: only doctest relevant files

### PR DESCRIPTION
**Pull request by pjonsson**
Avoid collecting files under cubedash/
unless they need to be tested (=contain
lines that start with whitespace followed
by ">>>").

This aligns with datacube-core and
minimizes the risk of triggering
deprecation warnings.

----
📚 Documentation preview 📚: https://datacube-explorer--801.org.readthedocs.build/en/801/

### CODE REVIEW

File: cubedash/conftest.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# This file is part of the Open Data Cube, see https://opendatacube.org for more information
+#
+# Copyright (c) 2015-2025 ODC Contributors
+# SPDX-License-Identifier: Apache-2.0
+from pathlib import Path
+
+
+# The only thing tested under this directory is doctests, so only collect
+# files that need testing (=have lines starting with whitespace + ">>>").
+def pytest_ignore_collect(collection_path: Path) -> bool:
```
Comment (omad) [inline cubedash/conftest.py:13]:
```suggestion
#
# This isn't a performance consideration, but is to 
# avoid triggering spurious deprecation warnings
def pytest_ignore_collect(collection_path: Path) -> bool:
```

File: MANIFEST.in [hunk 2]
```diff
@@ -1,4 +1,5 @@
 include cubedash/_version.py
+exclude cubedash/conftest.py
```
Comment (omad) [inline MANIFEST.in:2]: _sigh_, still using `setuptools` here. 😄

**Review by omad — APPROVED** [review]

File: MANIFEST.in (hunk 2 above)
Comment (pjonsson) [reply MANIFEST.in:2]: @omad I think almost all ODC repositories do!

But just being able to use uv has been a big win, so much easier to just do a quick `uv run ruff check` no matter what Ruff version a repository happens to be using currently.
